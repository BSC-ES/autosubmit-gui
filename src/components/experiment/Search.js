import React, { useState, useContext, Fragment } from 'react';
import ExperimentContext from '../context/experiment/experimentContext';
import AlertContext from '../context/alert/alertContext';

const Search = ({ setAlert }) => {
  const experimentContext = useContext(ExperimentContext);
  const alertContext = useContext(AlertContext);

  const [text, setText] = useState('');

  const onSubmit = e => {
    e.preventDefault();
    if (text === '') {
      alertContext.setAlert('Please enter something', 'light');
    } else {
      experimentContext.searchExperiments(text);
      //setText('');
    }
  };

  const onChange = e => setText(e.target.value);

  return (
    <div className="container">
      <form onSubmit={onSubmit} className='form'>
        <div className="input-group mb-3">
          
            <input
              className="form-control"
              type='text'
              name='text'
              placeholder='Search Experiments by Expid or Description...'
              value={text}
              onChange={onChange}
            />
            <div className="input-group-append">
              <input          
                type='submit'
                value='Search'
                className='btn btn-dark'
              />
            </div>        
        </div>
      </form>
      {experimentContext.experiments.length > 0 && (
            <Fragment>
              <button className='btn btn-light btn-block' onClick={experimentContext.clearExperiments}>
                Clear
              </button>
              <br></br>
            </Fragment>                                  
          )} 
    </div>
  );
}

export default Search;
