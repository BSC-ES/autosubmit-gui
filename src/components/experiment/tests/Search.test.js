import React from "react";
import { unmountComponentAtNode, render } from "react-dom";
import { BrowserRouter as Router } from "react-router-dom";
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';
import Search from "../Search";
import ExperimentContext from "../../context/experiment/experimentContext";
import AlertContext from "../../context/alert/alertContext";

let container = null;
// const experiment = {branch: "master", description: "Test", owner:"wuruchi", owner_id: 1226, hpc:"LOCAL", db_historic_version: 14, version:"3.11.0"};

const experiments = {
  "experiment": [
      {
          "completed": 0,
          "description": "TEST CASE MN3 auto-ecearth3 layer 2: ORCA1L75 (oce_nudg)",
          "hpc": "marenostrum3",
          "id": 1228,
          "name": "t012",
          "status": "NOT RUNNING",
          "total": 0,
          "user": "mcastril",
          "version": "3.7.3"
      },
      {
          "completed": "NA",
          "description": "test1 auto-ml4aq",
          "hpc": "power9",
          "id": 4758,
          "name": "a2jp",
          "status": "NOT RUNNING",
          "total": "NA",
          "user": "mcastril",
          "version": "3.11.0-hotfix"
      },
      {
          "completed": 0,
          "description": "SRun Vs Threads",
          "hpc": "marenostrum4",
          "id": 4852,
          "name": "a2lf",
          "status": "NOT RUNNING",
          "total": 19,
          "user": "mcastril",
          "version": "3.12.1b"
      },
      {
          "completed": 243,
          "description": "Copy of a2k5 with vertical-horizontal wrappers",
          "hpc": "marenostrum4",
          "id": 4896,
          "name": "a2m3",
          "status": "NOT RUNNING",
          "total": 243,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 51,
          "description": "Copy of a2k5",
          "hpc": "marenostrum4",
          "id": 5150,
          "name": "a2rj",
          "status": "NOT RUNNING",
          "total": 51,
          "user": "mcastril",
          "version": "3.12.1b0"
      },
      {
          "completed": 4,
          "description": "Copy of t0bz to do memory debugging",
          "hpc": "marenostrum4",
          "id": 5248,
          "name": "a2tk",
          "status": "NOT RUNNING",
          "total": 8,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 4,
          "description": "Copy of t0c6 to investigate ORCA025-standalone performance in EC-Earth3",
          "hpc": "marenostrum4",
          "id": 5331,
          "name": "a2vp",
          "status": "NOT RUNNING",
          "total": 5,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 0,
          "description": "X11 test",
          "hpc": "marenostrum4",
          "id": 5405,
          "name": "a2wz",
          "status": "NOT RUNNING",
          "total": 1,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 5,
          "description": "Copy of t0bz to check flags impact in performance",
          "hpc": "marenostrum4",
          "id": 5518,
          "name": "a2yp",
          "status": "NOT RUNNING",
          "total": 5,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 9,
          "description": "Copy of t0gg to measure Intel MPI performance",
          "hpc": "marenostrum4",
          "id": 5688,
          "name": "a31t",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 9,
          "description": "Copy of t0gg to measure Intel MPI performance",
          "hpc": "marenostrum4",
          "id": 5689,
          "name": "a31u",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 9,
          "description": "Copy of t0gg to measure Open MPI performance",
          "hpc": "marenostrum4",
          "id": 5690,
          "name": "a31v",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 9,
          "description": "Copy of t0gg to measure Intel MPI performance",
          "hpc": "marenostrum4",
          "id": 5691,
          "name": "a31w",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 9,
          "description": "Copy of t0gg to measure Intel MPI performance",
          "hpc": "marenostrum4",
          "id": 5692,
          "name": "a31x",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 9,
          "description": "Copy of t0gg to measure Open MPI performance",
          "hpc": "marenostrum4",
          "id": 5693,
          "name": "a31y",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 4,
          "description": "Copy of a31t to measure Intel MPI performance with OFI",
          "hpc": "marenostrum4",
          "id": 5697,
          "name": "a322",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 4,
          "description": "Copy of a31t to check EC-Earth working with different number of cores",
          "hpc": "marenostrum4",
          "id": 5701,
          "name": "a325",
          "status": "NOT RUNNING",
          "total": 9,
          "user": "mcastril",
          "version": "3.13.0b0"
      },
      {
          "completed": 32,
          "description": "Copy of a34m (t0go) experiment to analyse wrappers performance",
          "hpc": "marenostrum4",
          "id": 5835,
          "name": "a34n",
          "status": "NOT RUNNING",
          "total": 147,
          "user": "mcastril",
          "version": "3.12.0"
      },
      {
          "completed": 0,
          "description": "Experiment to start debugging Auto-NEMO",
          "hpc": "marenostrum4",
          "id": 5976,
          "name": "a37h",
          "status": "NOT RUNNING",
          "total": 21,
          "user": "mcastril",
          "version": "3.13.0b0"
      }
  ]
};

beforeEach(() => {
  // setup DOM element as render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("Search renders with content", () => {
  
  act(() => {
    render(<Router><ExperimentContext.Provider value={{ experiments: experiments, searchExperiments: () => null, getCurrentRunning: () => null, experimentRunning:true, getSummaries: () => null, clearExperiments: () => null}}>
          <AlertContext.Provider 
          value={{ setAlert: () => null }}>
            <Search />
          </AlertContext.Provider>
      </ExperimentContext.Provider></Router>, container);
  });
  expect(container.innerHTML).toContain("Search");
  expect(container.innerHTML).toContain("Active Exps");
});

// --- Search.js ---

it("Search renders correctly", () => {
  const tree = renderer.create(<Router><ExperimentContext.Provider value={{ experiment: {}, experiments: [], summaries: new Map(), loadingSummary: new Map()}}><AlertContext.Provider value={{}}><Search/></AlertContext.Provider></ExperimentContext.Provider></Router>).toJSON();
  expect(tree).toMatchSnapshot();
})