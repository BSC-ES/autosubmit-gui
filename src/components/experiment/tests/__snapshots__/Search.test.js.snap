// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row-hl d-flex flex-wrap mb-2"
  >
    <div
      className="item-hl mr-1 px-2 pt-1 border rounded"
    >
      <div
        className="form-check form-check-inline"
      >
        <input
          checked={false}
          className="form-check-input"
          id="experimentTypeTest"
          name="experimentType"
          onChange={[Function]}
          type="radio"
          value="test"
        />
        <label
          className="form-check-label"
          htmlFor="experimentTypeTest"
        >
          Test
        </label>
      </div>
      <div
        className="form-check form-check-inline"
      >
        <input
          checked={false}
          className="form-check-input"
          id="experimentTypeExperiment"
          name="experimentType"
          onChange={[Function]}
          type="radio"
          value="experiment"
        />
        <label
          className="form-check-label"
          htmlFor="experimentTypeExperiment"
        >
          Experiment
        </label>
      </div>
      <div
        className="form-check form-check-inline"
      >
        <input
          checked={false}
          className="form-check-input"
          id="experimentTypeAll"
          name="experimentType"
          onChange={[Function]}
          type="radio"
          value="all"
        />
        <label
          className="form-check-label"
          htmlFor="experimentTypeAll"
        >
          All
        </label>
      </div>
    </div>
    <div
      className="item-hl mr-1 pt-1 px-2 border rounded"
    >
      <div
        className="form-check"
      >
        <input
          checked={false}
          className="form-check-input"
          id="switchActive"
          onChange={[Function]}
          type="checkbox"
          value="active"
        />
        <label
          className="form-check-label"
          htmlFor="switchActive"
        >
          Only Active
        </label>
      </div>
    </div>
    <div
      className="item-hl flex-fill mr-1"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            name="text"
            onChange={[Function]}
            placeholder="If it uses Autosubmit, you will find it. Search by expid, description, or owner."
            type="text"
            value=""
          />
          <div
            className="input-group-append"
          >
            <input
              className="btn btn-dark"
              type="submit"
              value="Search"
            />
          </div>
        </div>
      </form>
    </div>
    <div
      className="item-hl"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div
          className="input-group"
        >
          <input
            className="btn btn-success"
            data-placement="bottom"
            data-toggle="tooltip"
            title="Shows all the experiments that are currently running under Autosubmit."
            type="submit"
            value="Active Exps"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
